### ISSUE TITLE
[caffe emiter] separate pad_h and pad_w for conv. layer

### ISSUE THREAD

**Issue by seovchinnikov** [Microsoft/MMdnn#114]
Now there can be only w=h padding for  L.Convolution(...pad=?...).
I think it's crucial feature to separate padding for h and w.

```
 
  // For 2D convolution only, the *_h and *_w versions may also be used to
  // specify both spatial dimensions.
  optional uint32 pad_h = 9 [default = 0]; // The padding height (2D only)
  optional uint32 pad_w = 10 [default = 0]; // The padding width (2D only)
```

**Comment by seovchinnikov** [comment]
Also, I dont think that the current implementation 
https://github.com/Microsoft/MMdnn/blob/242edc9b2d3e64bc7eaf9e299eeb2ee52f865de1/mmdnn/conversion/caffe/caffe_emitter.py#L163

`IR_node.get_attr('pads')[1]`
is correct, because in the following example extracted from TF:

```
{
  "node": [
    {
      "attr": {
        "dtype": {
          "type": "DT_FLOAT32"
        },
        "_output_shapes": {
          "list": {
            "shape": [
              {
                "dim": [
                  {
                    "size": "1"
                  },
                  {
                    "size": "432"
                  },
                  {
                    "size": "368"
                  },
                  {
                    "size": "3"
                  }
                ]
              }
            ]
          }
        },
        "shape": {
          "shape": {
            "dim": [
              {
                "size": "1"
              },
              {
                "size": "432"
              },
              {
                "size": "368"
              },
              {
                "size": "3"
              }
            ]
          }
        }
      },
      "op": "DataInput",
      "name": "image"
    },
    {
      "input": [
        "image"
      ],
      "attr": {
        "strides": {
          "list": {
            "i": [
              "1",
              "2",
              "2",
              "1"
            ]
          }
        },
        "pads": {
          "list": {
            "i": [
              "0",
              "0",
              "0",
              "0",
              "0",
              "1",
              "1",
              "0"
            ]
          }
        },
        "_output_shapes": {
          "list": {
            "shape": [
              {
                "dim": [
                  {
                    "size": "1"
                  },
                  {
                    "size": "216"
                  },
                  {
                    "size": "184"
                  },
                  {
                    "size": "24"
                  }
                ]
              }
            ]
          }
        },
        "auto_pad": {
          "s": "U0FNRV9MT1dFUg=="
        },
        "kernel_shape": {
          "list": {
            "i": [
              "3",
              "3",
              "3",
              "24"
            ]
          }
        },
        "data_format": {
          "s": "TkhXQw=="
        }
      },
      "op": "Conv",
      "name": "MobilenetV1/Conv2d_0/Conv2D"
    }
}
```

IR_node.get_attr('pads')[1] would be 0 but the actual padding is 1

**Comment by kitstar** [comment]
Hi @seovchinnikov 
1. @TobeyQin can help to implement it.
1. The TensorFlow padding calculate algorithm is different from Caffe, if you read the APIs detail.

**Comment by TobeyQin** [comment]
resolved by #136
